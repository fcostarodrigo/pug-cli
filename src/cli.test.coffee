{ promises: { readFile, writeFile, mkdir, rmdir, readdir } } = require 'fs'
{ spawnSync } = require 'child_process'
{ join } = require 'path'

testFolder = 'cliTest'

script = join '..', 'src', 'cli.coffee'

beforeEach ->
  await mkdir testFolder
  process.chdir testFolder

afterEach ->
  process.chdir '..'
  await rmdir testFolder, { recursive: true }

test 'display help', ->
  process = spawnSync 'coffee', [script, '--help']
  expect(process.stdout.toString()).toMatchSnapshot();

test 'compile a pug file', ->
  await writeFile 'index.pug', 'div hello'

  spawnSync 'coffee', [script]

  content = await readFile 'index.html', 'utf-8'
  expect(content).toBe('<div>hello</div>')

test 'compile a specific pug file', ->
  await writeFile 'index1.pug', 'div hello'
  await writeFile 'index2.pug', 'div hello'

  spawnSync 'coffee', [script, 'index1.pug']

  files = await readdir '.'
  expect(files.sort()).toEqual [
    'index1.pug'
    'index2.pug'
    'index1.html'
  ].sort()

test 'compile to a target folder', ->
  await writeFile 'index.pug', 'div hello'

  spawnSync 'coffee', [script, '--out', 'dist']

  content = await readFile join('dist', 'index.html'), 'utf-8'
  expect(content).toBe('<div>hello</div>')
