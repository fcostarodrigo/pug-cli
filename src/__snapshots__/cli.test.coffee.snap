// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display help 1`] = `
"coffee [files..]

Positionals:
  files  Files and folders to compile                           [default: [\\".\\"]]

Options:
      --help                  Show help                                [boolean]
      --version               Show version number                      [boolean]
  -o, --out                   Output directory                          [string]
  -c, --config                Configuration file path                   [string]
  -i, --ignore                Path of file containing patterns like gitignore
                                                [string] [default: \\".pugignore\\"]
  -e, --extensions            List of extensions to compile
                                             [array] [default: [\\".jade\\",\\".pug\\"]]
      --options.basedir       The root directory of all absolute inclusion
                                                                        [string]
      --options.doctype       Doctype to include in the templates if not
                              specified                                 [string]
      --options.self          Use a self namespace to hold the locals  [boolean]
      --options.debug         Log tokens and functions to stdout       [boolean]
      --options.compileDebug  Include function source in the compiled template
                                                                       [boolean]
"
`;
